tests/foreman/ui/test_http_proxy.py

```diff
@@ -67,3 +82,117 @@ def test_positive_create_update_delete(session, module_org, module_loc):
+    # Create repositories from UI.
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        repo_a1_name = gen_string('alpha')
+        session.repository.create(
+            product_a.name,
+            {
+                'name': repo_a1_name,
+                'repo_type': REPO_TYPE['yum'],
+                'repo_content.upstream_url': FAKE_1_YUM_REPO,
+                'repo_content.http_proxy_policy': 'No HTTP Proxy',
+            },
+        )
+        repo_a1_values = session.repository.read(product_a.name, repo_a1_name)
+        assert repo_a1_values['repo_content']['http_proxy_policy'] == 'No HTTP Proxy'
+        repo_a2_name = gen_string('alpha')
+        session.repository.create(
+            product_a.name,
+            {
+                'name': repo_a2_name,
+                'repo_type': REPO_TYPE['yum'],
+                'repo_content.upstream_url': FAKE_1_YUM_REPO,
+                'repo_content.http_proxy_policy': 'Use specific HTTP Proxy',
+                'repo_content.proxy_policy.http_proxy': http_proxy_a.name,
+            },
+        )
+        repo_a2_values = session.repository.read(product_a.name, repo_a2_name)
+        assert repo_a2_values['repo_content'][
+            'http_proxy_policy'
+        ] == 'Use specific HTTP Proxy ({})'.format(http_proxy_a.name)
+        repo_b1_name = gen_string('alpha')
+        session.repository.create(
+            product_b.name,
+            {
+                'name': repo_b1_name,
+                'repo_type': REPO_TYPE['puppet'],
+                'repo_content.upstream_url': FAKE_0_PUPPET_REPO,
+                'repo_content.http_proxy_policy': 'Global Default',
+            },
+        )
+        repo_b1_values = session.repository.read(product_b.name, repo_b1_name)
+        assert 'Global Default' in repo_b1_values['repo_content']['http_proxy_policy']
+        repo_b2_name = gen_string('alpha')
+        session.repository.create(
+            product_b.name,
+            {
+                'name': repo_b2_name,
+                'repo_type': REPO_TYPE['puppet'],
+                'repo_content.upstream_url': FAKE_0_PUPPET_REPO,
+                'repo_content.http_proxy_policy': 'No HTTP Proxy',
+            },
+        )
+        # Add http_proxy to products
+        session.product.search('')
+        session.product.manage_http_proxy(
+            [product_a.name, product_b.name],
+            {
+                'http_proxy_policy': 'Use specific HTTP Proxy',
+                'proxy_policy.http_proxy': http_proxy_b.name,
+            },
+        )
+        # Verify that Http Proxy is updated for all repos of product_a and product_b.
+        repo_a1_values = session.repository.read(product_a.name, repo_a1_name)
+        assert repo_a1_values['repo_content'][
+            'http_proxy_policy'
+        ] == 'Use specific HTTP Proxy ({})'.format(http_proxy_b.name)
```

Comment:
There is a lot of string duplication here. Can you get away with following?
```python
        proxy_policy = 'Use specific HTTP Proxy ({})'
        assert repo_a1_values['repo_content']['http_proxy_policy'] == proxy_policy.format(http_proxy_b.name)
        repo_a2_values = session.repository.read(product_a.name, repo_a2_name)
        assert repo_a2_values['repo_content']['http_proxy_policy'] == proxy_policy.format(http_proxy_b.name)
        ...
```

Reply:
yeah, it doesn't look good. Updated PR.